### PR TITLE
Fix S2933 FP: Support unsigned right-shift operator (>>>)

### LINKED ISSUE #6313: Fix S2933 FP: Support unsigned right-shift operator (>>>)
### Description

FP for [S2933](https://sonarsource.github.io/rspec/#/rspec/S2933/csharp) (FieldShouldBeReadonly).

Using C#11 unsigned right-shift operator (>>>), the rule is raising an issue.

### Repro steps

```
class Person
{
    int somefield = 42; // Noncompliant FP (it is part of an UnsignedRightShiftAssignmentExpression)
    private readonly Action<int> setter;

    Person(int birthYear)
    {
        setter = i => { somefield >>>= i; };
    }
}
```

### Expected behavior

The rule should not raise an issue.

### Actual behavior

The rule raises an issue.

### PR DESCRIPTION
**Pull request by gregory-paidis-sonarsource**
Fixes #6313

### CODE REVIEW

**Review by cristian-ambrosini-sonarsource — APPROVED** [review]
Great!

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/FieldShouldBeReadonly.CSharp11.cs [hunk 1]
```diff
@@ -49,7 +49,7 @@ private void M()
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/FieldShouldBeReadonly.CSharp11.cs:49]: NITPICK: Line 28 is not necessary.

File: analyzers/src/SonarAnalyzer.CSharp/Rules/FieldShouldBeReadonly.cs [hunk 2]
```diff
@@ -57,7 +57,8 @@ public sealed class FieldShouldBeReadonly : SonarDiagnosticAnalyzer
             SyntaxKind.OrAssignmentExpression,
             SyntaxKind.LeftShiftAssignmentExpression,
             SyntaxKind.RightShiftAssignmentExpression,
-            SyntaxKindEx.CoalesceAssignmentExpression
+            SyntaxKindEx.CoalesceAssignmentExpression,
+            SyntaxKindEx.UnsignedRightShiftAssignmentExpression
         };
```
Comment (csaba-sagi-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/FieldShouldBeReadonly.cs:62]: Please make `ReadonlyFieldCollector` and `PartialTypeDeclarationProcessor` classes sealed.

File: analyzers/src/SonarAnalyzer.CSharp/Rules/FieldShouldBeReadonly.cs [hunk 3]
```diff
@@ -113,7 +114,7 @@ private class ReadonlyFieldCollector
         {
```
Comment (csaba-sagi-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/FieldShouldBeReadonly.cs:114]: On line 85 and 90. The two `if`s can be merged into one.

**Review by csaba-sagi-sonarsource — CHANGES_REQUESTED** [review]
Small cleanup changes requested.

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**